clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/gitRepo/GitRepoArtifactCredentials.java

```diff
@@ -122,8 +121,32 @@ public InputStream download(Artifact artifact) throws IOException {
     }
   }
 
+  private String makeGitReference(Git repo, String gitRef) throws GitAPIException {
+    String branchOrCommit;
+    String remoteRef;
+    Boolean branch = false;
+    if (gitRef.contains("/")) {
+      String[] path = gitRef.split("/");
+      branchOrCommit = path[path.length - 1];
+    } else {
+      branchOrCommit = gitRef;
+    }
+    String possibleRemoteRef = "refs/heads/" + branchOrCommit;
+    Collection<Ref> call = repo.lsRemote().setHeads(true).call();
```

Comment:
Would it be possible to get this information without making another call out to the remote? Is this data available when we do the initial clone?

Reply:
I agree, it'd be nice to keep from making another api call. I might be able to use `branchList` instead. I'm going to give that a try.